Subject: Linux-RT 2.6.24-rt1
From: http://www.kernel.org/pub/linux/kernel/projects/rt/
Acked-by: Sven-Thorsten Dietrich <sdietrich@suse.de>
From fad7c246ce9627cc06fa936f41b5cd40f5097f13 Mon Sep 17 00:00:00 2001
From: Dmitry Adamushko <dmitry.adamushko@gmail.com>
Date: Tue, 11 Dec 2007 10:02:47 +0100
Subject: [PATCH] sched: no need for 'affine wakeup' balancing in

No need to do a check for 'affine wakeup and passive balancing possibilities' in
select_task_rq_fair() when task_cpu(p) == this_cpu.

I guess, this part got missed upon introduction of per-sched_class select_task_rq()
in try_to_wake_up().

Signed-off-by: Dmitry Adamushko <dmitry.adamushko@gmail.com>
Signed-off-by: Ingo Molnar <mingo@elte.hu>

---
 kernel/sched_fair.c |    3 +++
 1 file changed, 3 insertions(+)

Index: linux-2.6.24-rt1/kernel/sched_fair.c
===================================================================
--- linux-2.6.24-rt1.orig/kernel/sched_fair.c	2008-01-25 15:06:58.000000000 -0500
+++ linux-2.6.24-rt1/kernel/sched_fair.c	2008-01-25 15:07:04.000000000 -0500
@@ -900,6 +900,9 @@ static int select_task_rq_fair(struct ta
 	this_cpu = smp_processor_id();
 	new_cpu  = cpu;
 
+	if (cpu == this_cpu)
+		goto out_set_cpu;
+
 	for_each_domain(this_cpu, sd) {
 		if (cpu_isset(cpu, sd->span)) {
 			this_sd = sd;
